packages/lesswrong/components/Layout.tsx

```diff
@@ -34,6 +34,7 @@ export const petrovBeforeTime = new DatabasePublicSetting<number>('petrov.before
 const petrovAfterTime = new DatabasePublicSetting<number>('petrov.afterTime', 0)
 
 import { Link } from '../lib/reactRouterWrapper';
+import { LoginPopoverContextProvider } from './hooks/useLoginPopoverContext';
```

Comment:
I added this context to make it easier to trigger the login popup for new users. We can now do:
```
const currentUser = useCurrentUser();
const {onLogin} = useLoginPopoverContextProvider();
const onClick = useCallback(() => {
  if (!currentUser) {
    onLogin();
    return;
  }
  // Some action that requires being logged in
}, [currentUser, onLogin]);
```
from any react component to show the popup